internal/metrics/circuitbreaker.go

```diff
@@ -0,0 +1,171 @@
+package metrics
```

Comment:
I'd prefer to keep the metrics implementation in `internal/circuit/` - e.g. `internal/circuit/circuit_metrics.go`. This matches the pattern for the existing metrics - e.g. `internal/engine/engine_metrics.go`.